projects/python-client/src/datapane/client/api/report/core.py

```diff
@@ -242,17 +253,54 @@ def __init__(
             blocks: Allows providing the document blocks as a single list
 
         Returns:
-            A `Report` document object that can be uploaded, saved, etc.
+            A `App` document object that can be uploaded, saved, etc.
 
         ..tip:: Blocks can be passed using either arg parameters or the `blocks` kwarg, e.g.
-          `dp.Report(plot, table)` or `dp.Report(blocks=[plot, table])`
+          `dp.App(plot, table)` or `dp.App(blocks=[plot, table])`
 
         ..tip:: Create a list first to hold your blocks to edit them dynamically, for instance when using Jupyter, and use the `blocks` parameter
         """
         super().__init__(**kwargs)
         self.page_layout = layout
         self._preprocess_pages(blocks or list(arg_blocks))
 
+    def upload(
+        self,
+        name: str,
+        description: str = "",
+        source_url: str = "",
+        publicly_visible: bool = False,
+        tags: t.List[str] = None,
+        project: t.Optional[str] = None,
+        open: bool = False,
+        formatting: t.Optional[AppFormatting] = None,
+        overwrite: bool = False,
+        **kwargs,
+    ) -> None:
+        warn(
+            "`App.upload` is deprecated and will be removed from future releases, use `Uploader` instead",
+            DeprecationWarning,
```

Comment:
for `DeprecationWarning`, I think we also set `stacklevel=2` so the warning refers to where the user called the deprecated code (see https://dev.to/hckjck/python-deprecation-2mof)